### ISSUE TITLE
[BUG] Page Cache does not get cleared correctly IMHO

### ISSUE THREAD

**Issue by viu-x** [a-r-m-i-n/fluid_page_cache#6]
Hi,

I love the idea of this extension and seems it it exactly what is needed in my current project. However, either I don't understand the clearing mechanism behind it correctly or it behaves wrong.

I have a sys folder (UID: 4) with several records of different types (all of the same extension and having relations with each other). These are display on a single page (UID: 7) either as a list, a grouped view or a single detail view. The page on which they get displayed, correctly gets all cache tags automatically. Amazing!

Then comes the problem. When I **update** any of those records nothing gets cleared. I checked with a debugger the following code:

https://github.com/a-r-m-i-n/fluid_page_cache/blob/5280540c23c2fd4d82af06ac398941452d6c7eca/Classes/Hooks/DataHandlerHook.php#L28-L33

The problem is that `$params['uid_page']` is **always** set to the UID of the sys folder in where my record are stored. Therefore it is never possible to clear only the page cache variations which contain a cache tag of the current edited table + UID.

In https://github.com/TYPO3/typo3/blob/11.5/typo3/sysext/core/Classes/DataHandling/DataHandler.php#L8904 that value is always set. So I have no idea how your can reach line 33 of your DataHandler hook.

If I however activate `tx_fluidpagecache_pid_cache_tag` on page 7 then obviously **all** page variations get cleared.

I don't know if I'm missing something or if it's a bug.

What I would expect/like of your extension:
- Fine grained page cache tags for each page (that's done amazingly)
- If I **update** an existing record, delete everything from page cache where the combination of table + uid exists as cache_tag (I think your extension wants to do that, but somehow it doesn't)
- Only purge the complete cache of a page with the cache tag `pid_X` if a **new** record is created.

As I said I don't know if I am missing something. It's my first time using this extension. Feel free to ask for further information if needed.

Best regards
Tobi

**Comment by ste101** [comment]
Take a look at #1

**Comment by viu-x** [comment]
Ok, so is there any plan to include the patch into your extension in the future?

**Comment by a-r-m-i-n** [comment]
Yes. No plan yet, when... But I would like to fix that. Probably when implementing v12 compatibility.

**Comment by mai-space** [comment]
Interesting. I have tested the configuration provided in the comment from @ste101.
Looks like he was right. In the previous implementation, all page alterations were cleared if only a record in the main page was changed, even if they are not on other page variations.

With the commit in bugfix/enhance-clear-cache-behaviour the behaviour was changed and now, the page variation stays as long as records on that variation or the page itself does not change.

Further testing with the viewhelper is needed but for now this seams like a valid bugfix.
